core/unit_test/CMakeLists.txt

```diff
@@ -90,6 +90,15 @@ KOKKOS_ADD_EXECUTABLE(
   ${COMPILE_ONLY_SOURCES}
 )
 
+if (WIN32)
+  KOKKOS_ADD_EXECUTABLE(
+    TestCompileOnlyWindowsInclude
+    SOURCES
+    TestWindowsInclude.cpp
+    TestCompileMain.cpp
+    )
+endif (WIN32)
```

Comment:
Just conditionally append it to `COMPILE_ONLY_SOURCES` above
(and please ~~`endif (WIN32)`~~ -> `endif()`

Reply:
I think this will be very brittle. Reasoning is the following: the other tests include headers that are not `Kokkos_Core.hpp` and thus do not have the guard against the macro problem. We can either guard every single one of them with the same macro push or we stick to the "we fix it for including `Kokkos_Core.hpp`" ... in which case we should add the test as a new executable to not break the other tests.

Reply:
You misunderstood, I meant do
```CMake
if (WIN32)
  list(APPEND COMPILE_ONLY_SOURCES TestWindowsInclude.cpp)
endif()
```
There is no point on adding a new **executable** (adding a new source file was the right thing of course)

Reply:
done in [1a0977](https://github.com/kokkos/kokkos/pull/5541/commits/1a0977bb7daf273b2298b75c009dc49cbded7a3d)